packages/python/src/armonik/common/helpers.py

```diff
@@ -96,3 +99,27 @@ def timedelta_to_duration(delta: timedelta) -> duration.Duration:
     d = duration.Duration()
     d.FromTimedelta(delta)
     return d
+
+
+def batched(iterable: Iterable[T], n: int) -> Iterable[List[T]]:
+    """
+    Batches elements from an iterable into lists of size at most 'n'.
+
+    Args:
+        iterable : The input iterable.
+        n : The batch size.
+
+    Yields:
+        A generator yielding batches of elements from the input iterable.
+    """
+    it = iter(iterable)
+    while True:
+        batch = []
+        try:
+            for i in range(n):
+                batch.append(next(it))
+        except StopIteration:
+            if len(batch) > 0:
+                yield batch
+            break
+        yield batch
```

Comment:
usually, I don't like while True and try catch in this kind of functions. Any better solution ?

Reply:
```suggestion
    sentinel = object()
    batch = []
    c = next(it, sentinel)
    while c is not sentinel:
        batch.append(c)
        if len(batch) == n:
            yield batch
            batch.clear()
        c = next(it, sentinel)
    if len(batch) > 0:
        yield batch
```